src/AstGen.zig

```diff
@@ -7303,6 +7303,7 @@ fn builtinCall(
             const mulend2 = try expr(gz, scope, .{ .ty = float_type }, params[2]);
             const addend = try expr(gz, scope, .{ .ty = float_type }, params[3]);
             const result = try gz.addPlNode(.mul_add, node, Zir.Inst.MulAdd{
+                .ty = float_type,
```

Comment:
no need to add the float type here- all 3 operands have been coerced to it so Sema can get it from there.